ssg/checks.py

```diff
@@ -57,6 +57,33 @@ def is_cce_valid(cceid):
     return match is not None
 
 
+def is_cce_value_valid(cceid):
+    # For context, see:
+    # https://github.com/ComplianceAsCode/content/issues/3044#issuecomment-420844095
+
+    # concat(substr ... , substr ...) -- just remove non-digit characters.
+    # Since we've already validated format, this hack suffices:
+    cce = re.sub(r'(CCE|-)', '', cceid)
+
+    # (string-to-codepoints - 48) is the same as int. Add in the reverse too.
+    # Double list cast is necessary: map isn't reversible, and generator
+    # isn't indexable.
+    digits = list(reversed(list(map(int, cce))))
+
+    # Even indices are doubled. Coerce to list for list addition. However,
+    # XPath uses 1-indexing so "evens" and "odds" are swapped from Python.
+    evens = list(map(lambda i: i*2, digits[1::2]))
+    odds = digits[0::2]
+
+    # XPath flattens the for loop's return into a single list; do this in
+    # via a double sum.
+    tuples = map(lambda j: [j % 10, j // 10], evens + odds)
+    value = sum(sum(tuples, [])) % 10
+
+    # Valid CCE <=> value == 0
+    return value == 0
+
```

Comment:
This looks more complex than necessary. 
This is the check algorithm, https://en.wikipedia.org/wiki/Luhn_algorithm.
Is there a special reason to do it exactly like it is done in XPath script?

Reply:
Yes, because it is how NIST does it in the validation program, which in this case, is the official documentation of how to calculate and validate CCE. 

Have you been able to find where the NIST defines how to calculate CCE such as explicit use of the Luhn algorithm?

Reply:
OMG.... I actually found the reference page for CCE that does specify Luhn.... @yuumasato you are amazing to find that! Super kudos to you.

@cipherboy can you update the PR to use https://en.wikipedia.org/wiki/Luhn_algorithm ?

Reply:
Had fond this: https://cce.mitre.org/lists/entries.html
But looking under NIST, found this: https://csrc.nist.gov/Projects/Security-Content-Automation-Protocol/faqs 
Look in " What is the format of the CCE Identifier (CCE-ID) number?"

Reply:
@yuumasato yeah... I had no idea why I couldn't find it. Great job!

Reply:
It was not easy to find it, :P